src/test/daml/Daml/Finance/Instrument/Bond/Test/Util.daml

```diff
@@ -138,58 +129,3 @@ lifecycleAndVerifyBondPaymentEffects readAs today instrument settlers issuer inv
 
   pure newBondInstrumentKey
-
--- | Verify a coupon payment of a bond (excluding the last date, which also includes a redemption amount).
--- Since we have moved to unit tests, this function is mainly kept for the docs.
--- It contains the code snippets needed for the bond lifecycling tutorial.
-lifecycleAndVerifyCouponEffectsAndSettlement : [Party] -> Date -> Instrument.K -> Parties -> Party -> Party -> ContractId Transferable.I -> ContractId Transferable.I -> AccountKey -> AccountKey -> [(Text, Parties)] -> Party -> [ContractId NumericObservable.I] -> Script (Instrument.K, ContractId Transferable.I)
-lifecycleAndVerifyCouponEffectsAndSettlement readAs today instrument settlers issuer investor investorBondTransferableCid custodianCashTransferableCid custodianAccount investorAccount obs custodian observableCids = do
-  (lifecycleCid, [effectCid]) <- lifecycleInstrument readAs today instrument settlers issuer observableCids
-
-  newBondInstrumentKey <- submit issuer do
-    Instrument.toKey <$> exerciseCmd @Instrument.I (coerceContractId lifecycleCid) Instrument.GetView with viewer = issuer
-
-  -- CREATE_SETTLEMENT_FACTORY_BOND_BEGIN
-  -- Create settlement factory
-  settlementFactoryCid <- toInterfaceContractId <$> submit investor do createCmd Factory with provider = investor; observers = empty
-  -- CREATE_SETTLEMENT_FACTORY_BOND_END
-
-  -- CLAIM_EFFECT_BOND_BEGIN
-  -- Claim effect
-  lifecycleClaimRuleCid <- toInterfaceContractId @Claim.I <$> submitMulti [custodian, investor] [] do
-    createCmd Claim.Rule
-      with
-        providers = fromList [custodian, investor]
-        claimers = singleton investor
-        settlers
-        settlementFactoryCid
-
-  result <- submitMulti [investor] readAs do
-    exerciseCmd lifecycleClaimRuleCid Claim.ClaimEffect with
-      claimer = investor
-      holdingCids = [toInterfaceContractId @Base.I investorBondTransferableCid]
-      effectCid
-      batchId = Id "CouponSettlement"
-  -- CLAIM_EFFECT_BOND_END
-
-  -- ALLOCATE_APPROVE_SETTLE_INSTRUCTIONS_BOND_BEGIN
-  let
-    Some [investorBondHoldingCid] = result.newInstrumentHoldingCids
-    [custodianCashInstructionCid] = result.instructionCids
-
-  -- Allocate instructions
-  (custodianCashInstructionCid, _) <- submitMulti [custodian] readAs do exerciseCmd custodianCashInstructionCid Instruction.Allocate with actors = singleton custodian; allocation = Pledge custodianCashTransferableCid
-
-  -- Approve instructions
-  custodianCashInstructionCid <- submit investor do
-    exerciseCmd custodianCashInstructionCid Instruction.Approve with actors = singleton investor; approval = TakeDelivery investorAccount
-
-  -- Settle batch
-  let settlerUsed = head $ toList settlers
-  [investorCashTransferableCid] <- submitMulti [settlerUsed] readAs do exerciseCmd result.batchCid Batch.Settle with actors = singleton settlerUsed
-  -- ALLOCATE_APPROVE_SETTLE_INSTRUCTIONS_BOND_END
-
```

Comment:
I think these code snippets are needed for the bond lifecycling tutorial. Currently, this function is not really used, it is just there to make sure that the code for the tutorial is covered by CI (if entered directly in the docs I think it will quickly be outdated).

Neither solution is obviously ideal. Maybe we should think about whether we need this tutorial at all, given other lifecycling documentation, and if so, what should be in it.
Generally, I would favour code that is part of a test over separate tutorial projects, from a maintainability point of view.

Reply:
Indeed, that's what I am now fixing in the docs 😅 for now I add a code snippet to the tutorial

I think that each tutorial should come with runnable code, so I will open an issue to replace the bond lifecycling tutorial with something for which we have a runnable test (e.g. dividend payment tutorial)

I removed the function because I had to spend some time updating it because of my changes, only to then find out that it wasn't being used anywhere

Reply:
Doc build is fixed and issue is tracked under https://github.com/digital-asset/daml-finance/issues/291